Hyper/Scene.cs

```diff
@@ -33,10 +33,19 @@ internal class Scene : IInputSubscriber
 
     public readonly SimulationManager<PoseIntegratorCallbacks> SimulationManager;
 
-    public readonly float Scale = 0.1f;
+    public readonly float Scale;
 
-    public Scene(float aspectRatio, float elevation, Context context)
+    public readonly Vector3i[] SphereCenters;
+
+    public readonly Vector3 LowerSphereCenter;
+
+    public Scene(Camera camera, float globalScale, float elevation, Context context)
     {
+        Scale = globalScale;
+        var sphere0Center = new Vector3i(0, 0, 0);
+        var sphere1Center = new Vector3i((int)(MathF.PI / Scale), 0, 0);
+        SphereCenters = new Vector3i[] { sphere0Center, sphere1Center };
+        LowerSphereCenter = new Vector3(sphere1Center.X, sphere1Center.Y, sphere1Center.Z) * Scale;
```

Comment:
I really hate having sphere things here even if i'ts not spherical. For the things that are only used in shaders can we just add them to the shaders and not have them in any classes? So we have seperate shaders for spherical. And if other classes need them can we add them in constructors of classes that need them?

Reply:
This is a very good point, I'll change it. I guess that a general rule of thumb would go like this: a class shouldn't have a field unless it directly uses it, and likewise a function shouldn't take an argument if it doesn't use it directly.